app/src/common/shared/org/mozilla/vrbrowser/browser/engine/Session.java

```diff
@@ -472,9 +472,9 @@ private void restore() {
 
         // We call restore when a session is first activated and when it's recreated.
         // We only need to notify of the session creation if it's not a recreation.
-        if (mSessionRemoved) {
+        if (addSession) {
             mSessionChangeListeners.forEach(listener -> listener.onSessionAdded(this));
-            mSessionRemoved = false;
+            mSessionAdded = true;
         }
```

Comment:
I don't understand why `mSessionAdded` can't be used. It would imply it is out of sync.